modules/programs/browserpass.nix

```diff
@@ -0,0 +1,90 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+with import ../lib/dag.nix;
+
+let
+  browsers = {
+    chrome = 1;
+    chromium = 2;
+    firefox = 3;
+    vivaldi = 4;
+  };
+in {
+  options = {
+    programs.browserpass = {
+      enable = mkEnableOption "the browserpass extension host application";
+
+      browsers = mkOption {
+        type = types.listOf types.str;
+        default = builtins.attrNames browsers;
+        example = [ "firefox" ];
+        description = "Which browsers to install browserpass for";
+      };
+    };
+  };
+
+  config = mkIf config.programs.browserpass.enable {
+    assertions = [
+      {
+        assertion = pkgs.stdenv.is64bit;
+        message = "Only 64-bit is supported";
+      }
+      {
+        assertion = builtins.all (x: builtins.any (y: x == y) (builtins.attrNames browsers)) config.programs.browserpass.browsers;
+        message = "Unsupported browser, must be one of: ${toString (builtins.attrNames browsers)}";
+      }
+      {
+        assertion = (builtins.getEnv "USER") != "root";
+        message = "root install not supported";
+      }
+    ];
+
+    home.activation.browserpass = let
+      browserpass = pkgs.stdenv.mkDerivation (let
```

Comment:
There seem to be a [browserpass](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/security/browserpass/default.nix) package in Nixpkgs trunk. It might be possible to back-port it to 17.03.

But I guess it cannot be uses as-is because of the permissions issue that you fix below.